cli/config_resolver.go

```diff
@@ -3,55 +3,56 @@ package cli
 import (
 	"io/ioutil"
 	"os"
+	"regexp"
+	"strings"
 
-	"github.com/alecthomas/kingpin"
+	"gopkg.in/alecthomas/kingpin.v2"
 	"gopkg.in/yaml.v2"
 )
 
-type configResolver []map[string]string
-
-func newConfigResolver() (configResolver, error) {
-	files := []string{
-		os.ExpandEnv("$HOME/.config/amtool/config.yml"),
-		"/etc/amtool/config.yml",
+var (
+	// kingpin transforms flags to <APPNAME>_<FLAG>.
+	envarTransformRegexp = regexp.MustCompile(`[^a-zA-Z0-9_]+`)
+	oldFlags             = map[string]string{
+		"comment_required": "require-comment",
 	}
+)
 
-	resolver := configResolver{}
+// loadConfig loads flags from YAML files and injects them as environment variables.
```

Comment:
Could any of these be secrets? The environment is not a safe place for secrets.

Reply:
An idea would be to store this in a public map at init time and use it to populate defaults for flags.. but maybe it is too convoluted

Reply:
The alertmanager.url can potentially contain basic auth credentials.

Any ideas on how else to implement this? @TheTincho's map idea might work. Parse the file and populate a struct/map, and set those as the default values for their appropriate flags?

Reply:
I've just tried and it doesn't work because of `alertmanager.url` (at least) which is a required flag . As stated in the kingpin [documentation](https://godoc.org/gopkg.in/alecthomas/kingpin.v2#FlagClause.Required): `You can not provide a Default() value to a Required() flag`.

I've tried the suggestion from https://github.com/alecthomas/kingpin/issues/208#issuecomment-327407428 but it doesn't work either because some flags are only valid for some commands and you can't have a flag present both in the configuration file and in the program's arguments (unless the flag can be repeated).

Reply:
Could we move it from being a `Required` flag, provide the value in the config file, and validate after the fact ourselves?

Reply:
@stuartnelson3 good idea, it seems to work...